adapters/aws-s3.md

```diff
@@ -0,0 +1,39 @@
+# AWS Simple Storage Service CloudEvents Adapter
+
+This document describes how to convert
+[AWS S3 events](https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html)
+into a CloudEvents.
+
+AWS S3 event documentation:
+https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html
+
+All S3 events are converted into CloudEvents using the
+same pattern as described in the following table:
+
+| CloudEvents Attribute | Value                                           |
+| :-------------------- | :---------------------------------------------- |
+| `id`                  | Generate a new unique value, e.g. a UUID        |
+| `source`              | "eventSource" value + `.` + "awsRegion" value + `.` + "s3.buckets.name" value  |
+| `specversion`         | `0.4-wip`                                       |
+| `type`                | `com.amazonaws.s3.` + "eventName" value         |
+| `datacontentencoding` | Omit                                            |
+| `datacontenttype`     | "Content-Type" value                            |
+| `schemaurl`           | Omit                                            |
+| `subject`             | "s3.object.key" value                           |
+| `time`                | "eventTime" value                               |
+| `data`                | Content of HTTP request body                    |
```

Comment:
What does `HTTP request body` refer to? The original S3 event?

S3 events aren't delivered via HTTP, they're put into SQS, SNS or invoke a Lambda function: https://docs.aws.amazon.com/AmazonS3/latest/dev/NotificationHowTo.html#how-to-enable-disable-notification-intro

Reply:
ok I think i fixed this.